File: 2026/05-May/31-Sunday/solution.js
var asteroidsDestroyed = function(mass, asteroids) {
    asteroids.sort((a, b) => a - b);

    let currentMass = mass;

    for (const asteroid of asteroids) {
        if (currentMass < asteroid) {
            return false;
        }
        currentMass += asteroid;
    }

    return true;
};

File: 2026/05-May/31-Sunday/solution.kt
class Solution {
    fun asteroidsDestroyed(mass: Int, asteroids: IntArray): Boolean {
        asteroids.sort()

        var currentMass = mass.toLong()

        for (asteroid in asteroids) {
            if (currentMass < asteroid) {
                return false
            }
            currentMass += asteroid
        }

        return true
    }
}
